pkg/controller/provider/web/ocp/namespace.go

```diff
@@ -97,12 +104,33 @@ func (h NamespaceHandler) Link(p *api.Provider, m *model.Namespace) string {
 	return h.Handler.Link(
 		NamespaceRoot,
 		base.Params{
-			base.NsParam:       p.Namespace,
-			base.ProviderParam: p.Name,
-			Ns2Param:           m.Name,
+			base.ProviderParam: string(p.UID),
+			NsParam:            m.PK,
 		})
 }
 
+//
+// Watch.
+func (h NamespaceHandler) watch(ctx *gin.Context) {
+	db := h.Reconciler.DB()
+	err := h.Watch(
+		ctx,
+		db,
+		&model.Namespace{},
+		func(in libmodel.Model) (r interface{}) {
+			m := in.(*model.Namespace)
+			vm := &Namespace{}
+			vm.With(m)
+			vm.SelfLink = h.Link(h.Provider, m)
+			r = vm
+			return
```

Comment:
This _builder_ `func` needs to be passed to the Watch().  The builder is used to convert the model object in the watch event into a REST resource.